consensus/consensus.go

```diff
@@ -1,103 +1,237 @@
 package consensus
 
 import (
-	"math/big"
+	"crypto/ecdsa"
+	"math"
 
 	log "github.com/Sirupsen/logrus"
 	"github.com/ubclaunchpad/cumulus/blockchain"
+	c "github.com/ubclaunchpad/cumulus/common/constants"
 )
 
-// MinedBlockCode is returned from ValidateMinedBlock
-type MinedBlockCode uint32
-
-const (
-	// ValidNewBlock is returned when a mined block is valid
-	ValidNewBlock MinedBlockCode = iota
-	// BadBlock is returned when an invalid block was mined
-	BadBlock MinedBlockCode = iota
-	// BadNonce is returned when the hash of the block is not less than the
-	// target
-	BadNonce MinedBlockCode = iota
-	// BadTarget is returned when the target that was used while mining is
-	// not equal to the current network target
-	BadTarget MinedBlockCode = iota
-	// BadBlockReward is returned when the block's rewards is not equal to the
-	// current reward of the network
-	BadBlockReward MinedBlockCode = iota
-	// BadCloudBase is returned when the address of the recipient of the
-	// CloudBase transaction is not valid
-	BadCloudBase MinedBlockCode = iota
-)
+// VerifyTransaction tests whether a transaction valid.
+func VerifyTransaction(bc *blockchain.BlockChain,
```

Comment:
This was my silly formatting using PEP8 on my go code from before 🤦‍♂️ . Feel free to undo it.